### PR TITLE
.service file removal on failure

### PR DESCRIPTION
**Pull request by baude**
Prodding bz #2024229 a little more, it turns out the service file is NOT
deleted when it is in a failed state (i.e the healtch check has failed
for some reason).  The state must be reset before the service is stopped
on container removal and then the files will be removed properly.

BZ#:2024229

[NO NEW TESTS NEEDED]

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/healthcheck_linux.go [hunk 1]
```diff
@@ -73,6 +73,14 @@ func (c *Container) removeTransientFiles(ctx context.Context) error {
 	defer conn.Close()
 	timerFile := fmt.Sprintf("%s.timer", c.ID())
 	serviceFile := fmt.Sprintf("%s.service", c.ID())
+
+	// If the service has failed (the healthcheck has failed), then
+	// the .service file is not removed on stopping the unit file. If
+	// we check the properties of the service, it will automatically
+	// reset the state.  But checking the state takes msecs vs usecs to
+	// blindly call reset and eat any errors
+	_ = conn.ResetFailedUnitContext(ctx, serviceFile)
```
Comment (rhatdan) [inline libpod/healthcheck_linux.go:82]: Should we at least logrus.Debug the errors?

Comment (baude) [reply libpod/healthcheck_linux.go:82]: I think it could be misleading, but ill do as reviewers want ...

Comment (rhatdan) [reply libpod/healthcheck_linux.go:82]: If it returns expected errors like ENOEXST, then don't log those.  I am worried about weird errors like EPERM, which we might not notice.

**Review by Luap99 — COMMENTED** [review]
LGTM

**Pull request merged**